### DISCUSSION TITLE
Parameter to navigate before or after getInitialProps / getServerSideProps

### DISCUSSION THREAD

**Discussion by slava-lu** [vercel/next.js#31614] [Ideas]
### Describe the feature you'd like to request

There are two major navigation strategies: 

1. First navigate with placeholders, then fetch data and re-render
2. First get the data, than navigate.
 
I see that recently the option 1 becomes more popular as it provides better user experience.  But pages with `getInitialProps / getServerSideProps` offer only option 2. 

To achieve option 1 behaviour we need to do a hack like this:

1. Check, if it is a server render or client navigation.
2. If client, skip the side effects in `getInitialProps / getServerSideProps` , return null props
3. Do side effects in `useEffect` hook.

### Describe the solution you'd like

It  would be good to have some kind of option for the Page to first render without the props returned via getInitialProps / getServerSideProps` and then re-render.

Some side notes related to this feature request:
 
There is no standard way to check if this is the first time server render or the client navigation.  For `getServerSideProps` community has come up with this weird hack  `const isServer = !req.url.startsWith('/_next')` or similar that checks for `/_next` in the url. Even if this feature looks big, some small thing for `isServer` would be nice.

It looks like that Vercel team is a bit obsessed with performance topic. Every their presentation revolves around it. But performance is not really everything and with the current network capacities and CPU powers is not really the hottest topic in Frontend development.  `getServerSideProps`  is absolutely unfriendly with `redux` and therefore with any bigger than a blog app development. It would be nice if some part of the team could focus on UI experience and global state management features as well.

**Comment by CristianCiubancan** [comment]
Awsome, I would love to see something like that implemented, it would be very useful along redux indeed. This would be the greatest christmas gift from vercel :)

**Comment by slava-lu** [comment]
some screenshot for vercel to have a look :)
![image](https://user-images.githubusercontent.com/32396113/142210433-c85ed46e-aece-471b-b90a-0e606cf83edf.png)

**Reply by leerob** [reply]
Hey! Did you see the Next.js 12 blog post? There’s an experimental flag for you to try out Suspense SSR today. 

https://nextjs.org/12
